impeller/entity/geometry/circle_geometry.cc

```diff
@@ -42,11 +41,12 @@ GeometryResult CircleGeometry::GetPositionBuffer(const ContentContext& renderer,
                                                  RenderPass& pass) const {
   auto& transform = entity.GetTransform();
 
-  Scalar half_width = stroke_width_ < 0
-                          ? 0.0
-                          : LineGeometry::ComputePixelHalfWidth(
-                                transform, stroke_width_,
-                                pass.GetSampleCount() == SampleCount::kCount4);
+  Scalar half_width = 0;
+  if (stroke_width_ > 0) {
+    auto [width, _] = LineGeometry::ComputePixelHalfWidth(
+        transform.GetMaxBasisLengthXY(), stroke_width_);
+    half_width = width;
+  }
```

Comment:
And if we don't pass it along to the stroke generator, does it generate an empty shape when tessellating?

Reply:
Is the comment just wrong? a quick read through and I think this code is doing reasonable things? half_width is always 0 for fills though its confusing that we go through the stroke code to get there...

Reply:
I think I see now, this is totally confusing. So the local field `stroke_width_` is set to -1 to indicate a filled circle. The `?:` code used to convert this to 0 and the half_width would always be >0 for the other case (stroked). So, yes, for the truly filled case we are calling a method called `StrokedCircle` which is odd and confusing, but the comment indicates that the called method will also do a filled circle for the case where the inner radius disappears - and it does no such thing.

Reply:
This code would make a lot more sense if there was a more explicit indicator of filled and it simply said:

```
if (filled) {
  generator = FilledCircle(...)
} else {
  half_width = ...;
  generator = StrokedCircle(..., half_width);
}
```

Fixing StrokedCircle to detect the degenerate case would be a separate fix, but until then, the comment below is just a red herring, maybe if it said "should simplify" instead of "will simplify"...

Reply:
Or, it would be OK if that pseudo-code used "stroke_width_ < 0" as the test, as long as it doesn't try to fold the cases together...

Reply:
I re-arranged this code.